tabbycat/users/permissions.py

```diff
@@ -0,0 +1,206 @@
+    VIEW_BREAK_CATEGORIES = 'view.breakcategories', _("view break categories")
+    VIEW_SPEAKER_CATEGORIES = 'view.speakercategories', _("view speaker categories")
+    EDIT_SPEAKER_CATEGORIES = 'edit.speakercategories', _("edit speaker categories")
+    VIEW_SPEAKER_ELIGIBILITY = 'view.speakereligibility', _("view speaker eligibility")
+    EDIT_SPEAKER_ELIGIBILITY = 'edit.speakereligibility', _("edit speaker eligibility")
+    VIEW_BREAK_OVERVIEW = 'view.break.overview', _("view break overview")
+    VIEW_BREAK = 'view.break', _("view breaks")
+    GENERATE_BREAK = 'generate.break', _("generate all breaks")
+
+    VIEW_PRIVATE_URLS = 'view.privateurls', _("view private urls")
+    VIEW_PRIVATE_URLS_EMAIL_LIST = 'view.privateurls.emaillist', _("view private urls email list")
+    GENERATE_PRIVATE_URLS = 'generate.privateurls', _("generate private URLs")
+    # need to get rid of generate private urls soons
+    SEND_PRIVATE_URLS = 'send.privateurls', _("send private URLs")
+
+    VIEW_CHECKIN = 'view.checkin', _("view checkins")
+    EDIT_PARTICIPANT_CHECKIN = 'edit.participantcheckin', _("edit participant check-in")
+    EDIT_ROOM_CHECKIN = 'edit.roomcheckin', _("edit room check-in")
+
+    EDIT_ROUND = 'edit.round', _("edit round attributes")
+    DELETE_ROUND = 'delete.round', _("delete rounds")
+    CREATE_ROUND = 'add.round', _("create rounds")
+
+    VIEW_EMAIL_STATUSES = 'view.emails', _("view email statuses")
+    SEND_EMAILS = 'send.emails', _("send participants email messages")
+
+    EXPORT_XML = 'export.xml', _("export DebateXML")
+
+    VIEW_SETTINGS = 'view.settings', _("view settings")
+    EDIT_SETTINGS = 'edit.settings', _("edit settings")
+
+
+permission_type = Union[Permission, bool]
+
+
+def has_permission(user: 'settings.AUTH_USER_MODEL', permission: permission_type, tournament: 'Tournament') -> bool:
+    if user.is_anonymous:
+        return False
+    if user.is_superuser:
+        return True
+
+    if isinstance(permission, bool):
+        return permission
+
+    if not hasattr(user, '_permissions'):
+        user._permissions = {}
+
+    if tournament.slug in user._permissions:
+        if permission in user._permissions[tournament.slug]:
+            return True
+    else:
+        user._permissions[tournament.slug] = set()
+
+    cached_perm = cache.get("user_%d_%s_%s_permission" % (user.pk, tournament.slug, str(permission)))
+    if cached_perm is not None:
+        if cached_perm:
+            user._permissions[tournament.slug].add(permission)
+        return cached_perm
+
+    perm = (
+        user.userpermission_set.filter(permission=permission, tournament=tournament).exists() or
+        user.membership_set.filter(group__permissions__contains=[permission], group__tournament=tournament).exists()
+    )
+    if perm:
+        user._permissions[tournament.slug].add(permission)
+        cache.set("user_%d_%s_%s_permission" % (user.pk, tournament.slug, str(permission)), perm)
```

Comment:
Just checking if this is supposed to fall back to the default timeout, which if I'm not mistaken is [5 minutes](https://docs.djangoproject.com/en/5.0/topics/cache/#cache-arguments)? This is fine imo, just wondering if you might've wanted to set the timeout here explicitly.